functions/src/compute-statistics.ts

```diff
@@ -24,25 +29,44 @@ export const computeStatistics = functions.https.onRequest(
       return;
     }
 
-    await firebaseTools.firestore.delete("stats", {
-      project: config.service_account.project_id,
-      recursive: true,
-      yes: true
-    });
-
     const votes = await db.collection("vote").get();
+    let paths: {
+      [key: string]: {
+        counters: { [key: string]: any };
+        path: FirebaseFirestore.DocumentReference;
+      };
+    };
+    paths = {};
     for (const vote of votes.docs) {
       const voteData = vote.data() as Vote;
-      await updateStats(
-        voteData.value,
-        vote.id,
-        firebase
-          .firestore()
-          .collection(`stats-campaign`)
-          .doc(vote.id),
-        { agency: voteData.agency }
-      );
+      const refsToAdd = getStatsRefsToUpdate(voteData);
+      for (const refToAdd of refsToAdd) {
+        if (paths[refToAdd.path]) {
+          paths[refToAdd.path] = {
+            counters: {
+              ...paths[refToAdd.path].counters,
+              ...{ agency: voteData.agency },
+              [voteData.value]:
+                (paths[refToAdd.path].counters[voteData.value] || 0) + 1
+            },
+            path: refToAdd
+          };
+        } else {
+          paths[refToAdd.path] = {
+            counters: {
+              ...{ agency: voteData.agency },
+              [voteData.value]: 1
+            },
+            path: refToAdd
+          };
+        }
+      }
```

Comment:
This is not clear enough for me. Suggestions for improvement:
- use a Map instead of an object to enable to use refs as keys
- `agency` should no hard coded here, use `getStatsRefsToUpdate` to compute both the ref and the additional fields store into it
- reduce the surface of the code that can mutate `paths` or make `paths` immutable